pkg/util/push/push.go

```diff
@@ -121,15 +137,59 @@ func handler(maxRecvMsgSize int,
 				level.Warn(logger).Log("msg", "push request canceled", "err", err)
 				return
 			}
+			var (
+				code int
+				msg  string
+			)
 			resp, ok := httpgrpc.HTTPResponseFromError(err)
-			if !ok {
-				http.Error(w, err.Error(), http.StatusInternalServerError)
-				return
+			if ok {
+				code, msg = int(resp.Code), string(resp.Body)
+			} else {
+				code, msg = getHTTPStatusAndMessage(err)
```

Comment:
```suggestion
				code, msg = distributorPushErrorHTTPStatus(err), err.Error()
```

And you can remove the `distributorPushErrorHTTPStatus` function.